### PR TITLE
New sample: Query features with arcade expression

### PR DESCRIPTION
**Pull request by duffh**
# Description

Added a new sample to demonstrate how to query features on a map using an arcade expression.

## Type of change

- [x] New sample implementation

**Platforms tested on**:

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] Xamarin.Forms Android
- [ ] Xamarin.Forms iOS
- [x] Xamarin.Forms UWP

## Checklist

- [x] Runs and compiles on all active platforms
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [ ] No unrelated changes have been made to any other code or project files
- [x] `sample_sync.py` runs without making changes
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] Code is commented with correct formatting
- [x] All variable and method names are good and make sense
- [x] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab

### CODE REVIEW

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            // Create an ArcGIS portal item.
+            var portal = await ArcGISPortal.CreateAsync();
+            var item = await PortalItem.CreateAsync(portal, "14562fced3474190b52d315bc19127f6");
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs:56]: We should try/catch portal creation.

File: src/Forms/Shared/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ namespace ArcGISRuntimeXamarin.Samples.EditFeatureLinkedAnnotation
         name: "Edit features with feature-linked annotation",
         category: "Data",
         description: "Edit feature attributes which are linked to annotation through an expression.",
-        instructions: "Pan and zoom the map to see that the text on the map is annotation, not labels. Tap one of the address points to update the house number .GetGeodatabaseAnnotationTable((AD_ADDRESS) and street name (ST_STR_NAM). Tap one of the dashed parcel polylines and tap another location to change its geometry. NOTE: Selection is only enabled for points and straight (single segment) polylines.",
+        instructions: "Pan and zoom the map to see that the text on the map is annotation, not labels. Tap one of the address points to update the house number (AD_ADDRESS) and street name (ST_STR_NAM). Tap one of the dashed parcel polylines and tap another location to change its geometry. NOTE: Selection is only enabled for points and straight (single segment) polylines.",
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml.cs:26]: Unrelated fix, should be in a seperate PR or commit.

File: src/Forms/Shared/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto" />
+            <RowDefinition Height="*" />
+        </Grid.RowDefinitions>
+        <ActivityIndicator x:Name="MyActivityIndicator"
+                           Grid.Row="0"
+                           IsVisible="False" />
+        <esriUI:MapView x:Name="MyMapView"
+                        Grid.Row="1"
+                        Margin="0,-6,0,0"
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml:15]: Margin values for the map view seem  weird.

File: src/Forms/Shared/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                // If the previously clicked feature is null or the previous feature ID does not match the current feature ID
+                // run the arcade expression query to get the crime count for a given feature.
+                if (_previousFeature == null || !(feature.Attributes["ID"].Equals(_previousFeature.Attributes["ID"])))
+                {
+                    // Show the activity indicator as the arcade evaluator evaluation call can take time to complete.
+                    MyActivityIndicator.IsVisible = true;
+                    MyActivityIndicator.IsRunning = true;
+
+                    // Instantiate a string containing the arcade expression.
+                    var expressionValue = "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs:92]:
```suggestion
                    string expressionValue = "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
```

File: src/WinUI/ArcGISRuntime.WinUI.Viewer/ArcGISRuntime.WinUI.Viewer.csproj [hunk 5]
```diff
@@ -24,7 +24,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <DefineConstants>WinUI</DefineConstants>
   </PropertyGroup>
-
```
Comment (ZackAllen) [inline src/WinUI/ArcGISRuntime.WinUI.Viewer/ArcGISRuntime.WinUI.Viewer.csproj:27]: Unrelated change

File: src/Forms/Shared/Forms.projitems [hunk 6]
```diff
@@ -10,6 +10,7 @@
   </PropertyGroup>
   <ItemGroup>
     <!-- Screenshots -->
+    <None Include="$(MSBuildThisFileDirectory)Samples\Data\QueryFeaturesWithArcadeExpression\readme.md" />
```
Comment (ZackAllen) [inline src/Forms/Shared/Forms.projitems:13]: There are incorrect breaking changes here.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿<UserControl x:Class="ArcGISRuntime.WPF.Samples.QueryFeaturesWithArcadeExpression.QueryFeaturesWithArcadeExpression"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:esri="http://schemas.esri.com/arcgis/runtime/2013">
+    <Grid>
+        <esri:MapView x:Name="MyMapView" GeoViewTapped="MyMapView_GeoViewTapped" />
+        <ProgressBar x:Name="ActivityIndicator"
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml:7]: This is really wide across the middle. It doesnt appear for very long on my system. Maybe we can make it smaller and move it off to a corner?

File: src/WinUI/ArcGISRuntime.WinUI.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<UserControl x:Class="ArcGISRuntime.WinUI.Samples.QueryFeaturesWithArcadeExpression.QueryFeaturesWithArcadeExpression"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+    <Grid>
+        <esriUI:MapView x:Name="MyMapView" GeoViewTapped="MyMapView_GeoViewTapped" />
+        <ProgressRing x:Name="ActivityIndicator"
```
Comment (ZackAllen) [inline src/WinUI/ArcGISRuntime.WinUI.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml:7]: I think we should also shove this in a corner.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/readme.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Create a `PortalItem` using the URL and ID.
+2. Create a `Map` using the portal item.
+3. Set the visibility of all the layers to false, except for the layer named "RPD Beats  - City_Beats_Border_1128-4500".
+4. Set up a lambda or a listener for clicks on the map.
+5. Identify the visible layer where it is tapped or clicked on and get the feature.
+6. Create an `ArcadeExpression` using the following string:
+
+	 ```	
+	 "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/readme.md:25]:
```suggestion
     "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/readme.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+## How it works
+
+1. Create a `PortalItem` using the URL and ID.
+2. Create a `Map` using the portal item.
+3. Set the visibility of all the layers to false, except for the layer named "RPD Beats  - City_Beats_Border_1128-4500".
+4. Set up a lambda or a listener for clicks on the map.
+5. Identify the visible layer where it is tapped or clicked on and get the feature.
+6. Create an `ArcadeExpression` using the following string:
+
+	 ```
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/readme.md:24]:
```suggestion
     ```	
```

**Review by ZackAllen — CHANGES_REQUESTED** [review]
I second the comments in review about clearing the callout when clicking outside of the beat boundaries.

File: src/Forms/Shared/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml.cs (hunk 2 above)
Comment (duffh) [reply src/Forms/Shared/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml.cs:26]: Good catch here.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            try
+            {
+                // Get the layer based on the position tapped on the MapView.
+                IdentifyLayerResult identifyResult = await MyMapView.IdentifyLayerAsync(_layer, e.Position, 12, false);
+
+                if (identifyResult == null || !identifyResult.GeoElements.Any())
+                {
+                    MyMapView.DismissCallout();
+
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs:88]: Unneeded space

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+                // If the previously clicked feature is null or the previous feature ID does not match the current feature ID
+                // run the arcade expression query to get the crime count for a given feature.
+                if (_previousFeature == null || !(feature.Attributes["ID"].Equals(_previousFeature.Attributes["ID"])))
+                {
+                    // Show the loading indicator as the arcade evaluator evaluation call can take time to complete.
+                    MyLoadingGrid.Visibility = Visibility.Visible;
+
+                    // Instantiate a string containing the arcade expression.
+                    string expressionValue = "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
+                                          "return Count(Intersects($feature, crimes));";
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml.cs:105]:
```suggestion
                                             "return Count(Intersects($feature, crimes));";
```
Might as well line it up.

File: src/WinUI/ArcGISRuntime.WinUI.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml [hunk 13]
```diff
@@ -0,0 +1,18 @@
+<UserControl x:Class="ArcGISRuntime.WinUI.Samples.QueryFeaturesWithArcadeExpression.QueryFeaturesWithArcadeExpression"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+    <Grid>
+        <esriUI:MapView x:Name="MyMapView" GeoViewTapped="MyMapView_GeoViewTapped" />
+        <Grid x:Name="MyLoadingGrid"
+              HorizontalAlignment="Stretch"
+              VerticalAlignment="Stretch"
+              Visibility="Visible">
```
Comment (ZackAllen) [inline src/WinUI/ArcGISRuntime.WinUI.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml:10]: Should not be visible at the start.

Comment (ZackAllen) [inline src/WinUI/ArcGISRuntime.WinUI.Viewer/Samples/Data/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.xaml:10]:
```suggestion
              Visibility="Collapsed">
```

**Review by ZackAllen — APPROVED** [review]
